cdap-api/src/main/java/io/cdap/cdap/api/macro/MacroEvaluator.java

```diff
@@ -46,11 +51,34 @@
 
   /**
    * Use the macro function and call the function with provided arguments,
-   * function uses the arguments and returns the evaluated response.
+   * function uses the arguments and returns the evaluated response as a string.
+   *
    * @param macroFunction macro function that has to be called
    * @param arguments arguments that will be passed to the macro function
    * @return value returned by macro function
    * @throws InvalidMacroException if macroFunction is not supported
    */
   String evaluate(String macroFunction, String... arguments) throws InvalidMacroException;
+
+  /**
+   * Use the macro function and call the function with provided arguments,
+   * function uses the arguments and returns the evaluated response as a map.
+   *
+   * @param macroFunction macro function that has to be called
+   * @param arguments arguments that will be passed to the macro function
+   * @return value returned by macro function
+   * @throws InvalidMacroException if macroFunction is not supported
+   */
+  default Map<String, String> evaluateMap(String macroFunction, String... arguments) throws InvalidMacroException {
+    throw new UnsupportedOperationException("Evaluating as map is not supported");
+  }
+
+  /**
+   * Get the list of macro function names that need to be evaluated as map.
+   *
+   * @return a set of functions that need to evaluated as map
+   */
+  default Set<String> getMapFunctions() {
```

Comment:
I think it would be a bit more extensible to have an API like:

```
  default ObjectType evaluateAs(String macroFunction)
```

where ObjectType is an enum with String and Map values right now. If we decide to add nested objects or something else in the future, we wouldn't need to introduce another getXFunctions() method.